libeos-updater-util/flatpak.c

```diff
@@ -1457,6 +1457,702 @@ euu_filter_for_existing_flatpak_ref_actions (GHashTable *ref_actions  /* (elemen
+                                                         NULL));
+
+  *out_remote_runtime_ref = g_steal_pointer (&runtime_remote_ref);
+  return TRUE;
+}
+
+static GPtrArray *
+list_related_refs_for_remote (FlatpakInstallation  *installation,
+                              const gchar          *remote_name,
+                              FlatpakRef           *source_ref,
+                              const gchar          *formatted_ref,
+                              GCancellable         *cancellable,
+                              GError              **error)
+{
+  g_autoptr(GPtrArray) remote_related_refs =
+    flatpak_installation_list_remote_related_refs_sync (installation,
+                                                        remote_name,
+                                                        formatted_ref,
+                                                        cancellable,
+                                                        error);
+  g_autoptr(FlatpakRemoteRef) remote_runtime_ref = NULL;
+  FlatpakRef *remote_runtime_ref_as_ref = NULL;
+
+  if (remote_related_refs == NULL)
+    return NULL;
+
+  /* Try and fetch the remote ref for the runtime, but if there isn't one
+   * then it is not an error */
+  if (!fetch_remote_runtime_ref (installation,
+                                 remote_name,
+                                 source_ref,
+                                 cancellable,
+                                 &remote_runtime_ref,
+                                 error))
+    return NULL;
+
+  if (remote_runtime_ref == NULL)
+    return g_steal_pointer (&remote_related_refs);
+
+  remote_runtime_ref_as_ref = FLATPAK_REF (remote_runtime_ref);
+  g_ptr_array_add (remote_related_refs, g_object_new (FLATPAK_TYPE_RELATED_REF,
+                                                      "kind", flatpak_ref_get_kind (remote_runtime_ref_as_ref),
+                                                      "name", flatpak_ref_get_name (remote_runtime_ref_as_ref),
+                                                      "arch", flatpak_ref_get_arch (remote_runtime_ref_as_ref),
+                                                      "branch", flatpak_ref_get_branch (remote_runtime_ref_as_ref),
+                                                      "should-delete", FALSE,
+                                                      "should-download", TRUE,
+                                                      NULL));
+  return g_steal_pointer (&remote_related_refs);
+}
+
+static gboolean
+populate_related_refs_in_all_remotes (FlatpakInstallation  *installation,
+                                      FlatpakRef           *source_ref,
+                                      GPtrArray            *related_refs_for_remotes,
+                                      GCancellable         *cancellable,
+                                      GError              **error)
+{
+  g_autofree gchar *formatted_ref = flatpak_ref_format_ref (source_ref);
+  guint i = 0;
```

Comment:
Nitpick: Now that #192 has been merged, you could actually move this into the `for` loop:
```
for (gsize i = 0; i < related_refs_for_remotes->len; i++)
```

How exciting!!